azure/scope/managedcontrolplane.go

```diff
@@ -570,10 +572,6 @@ func (s *ManagedControlPlaneScope) AgentPoolSpec() azure.AgentPoolSpec {
 		agentPoolSpec.MinCount = s.InfraMachinePool.Spec.Scaling.MinSize
 	}
 
-	if s.InfraMachinePool.Spec.MaxPods != nil {
-		agentPoolSpec.MaxPods = s.InfraMachinePool.Spec.MaxPods
```

Comment:
is the above equivalent because we expect MaxPods to never be nil because of defaulting (just double checking my understanding)?

Reply:
its actually just to simplify the code as if it's nil we are ok to pass nil. As both are `*int32`
e.g in case of other fields like `OSDiskSizeGB` the nil check makes sense as we are taking a `*int32` and passing it as `int32`